components/common/SEO/SEO.tsx

```diff
@@ -0,0 +1,129 @@
+import Head from 'next/head'
+import { FC } from 'react'
+import config from '@config/seo_meta.json'
+
+interface OgImage {
+  url?: string
+  width?: string
+  height?: string
+  alt?: string
+}
+
+interface Props {
+  title?: string
+  description?: string
+  robots?: string
+  openGraph?: {
+    title?: string
+    type?: string
+    locale?: string
+    description?: string
+    site_name?: string
+    url?: string
+    images?: OgImage[]
+  }
+}
+
+const ogImage = ({ url, width, height, alt }: OgImage, index: number) => {
+  return (
+    <>
+      <meta key={`og:image${index}`} property="og:image" content={url} />
+      <meta
+        key={`og:image:width${index}`}
+        property="og:image:width"
+        content={width}
+      />
+      <meta
+        key={`og:image:height${index}`}
+        property="og:image:height"
+        content={height}
+      />
+      <meta
+        key={`og:image:alt${index}`}
+        property="og:image:alt"
+        content={alt}
+      />
+    </>
+  )
+}
+
+const SEO: FC<Props> = ({ title, description, openGraph, robots }) => {
+  return (
+    <Head>
+      <title>
+        {title ? `${config.titleTemplate.replace(/%s/g, title)}` : config.title}
+      </title>
+      <meta
+        key="description"
+        name="description"
+        content={description || config.description}
+      />
+      <meta
+        key="og:type"
+        property="og:type"
+        content={openGraph?.title ?? config.openGraph.title}
+      />
+      <meta
+        key="og:title"
+        property="og:title"
+        content={
+          openGraph?.type ?? config.openGraph.type ?? title ?? config.title
+        }
+      />
+      <meta
+        property="og:description"
+        content={
+          openGraph?.description ??
+          config.openGraph.description ??
+          description ??
+          config.description
+        }
+      />
+      <meta
+        key="og:site_name"
+        property="og:site_name"
+        content={openGraph?.site_name ?? config.openGraph.site_name}
+      />
+      <meta
+        key="og:url"
+        property="og:url"
+        content={openGraph?.url ?? config.openGraph.url}
+      ></meta>
+      {openGraph?.locale && (
+        <meta key="og:locale" property="og:locale" content={openGraph.locale} />
+      )}
+      {openGraph?.images?.length
+        ? openGraph.images.map((img, index) => ogImage(img, index))
+        : ogImage(config.openGraph.images[0], 0)}
```

Comment:
I think that this should be a component instead of being called as a function

Reply:
I tried to do that as a JSX component but either it was not updating properly meta tags between the rerenders (https://github.com/vercel/commerce/pull/660/files#diff-ae719305395f324d42d061ca30f7e380b3383e94ccda8ce4b85dcfe7f4ed31dcR70) or I could return an array of `ReactNode[]` then TypeScript was complaining and failing build.